src/secp256k1/src/cpp/GroupElement.cpp

```diff
@@ -529,6 +529,10 @@ const unsigned char* GroupElement::deserialize(const unsigned char* buffer) {
     result.infinity = (int)infinity;
 
     secp256k1_gej_set_ge(reinterpret_cast<secp256k1_gej *>(g_), &result);
+
+    if (!secp256k1_ge_is_valid_var(&result)) {
+        throw "GroupElement: deserialize failed";
```

Comment:
Would be useful to have a unit test showing this works as expected.

Reply:
done